app/controllers/batches_controller.rb

```diff
@@ -0,0 +1,24 @@
+class BatchesController < ApplicationController
+    def index
+        if user_signed_in?
+            @batches = Batch.where(:user_id == current_user.id)
+        else
+            @batches = Batch.all
+        end
+        render json: @batches.to_json(:include => :tasks)
+    end
+
+    def create
+        batch = Batch.create(batch_params)
```

Comment:
Here too.  I’d do ```current_user.create_batch(batch_params)``` instead of passing along the user_id.  If you accept a user_id as a param, its a security risk.  I could create batches for you if I knew your user id, for example.